js/script.js

```diff
@@ -0,0 +1,33 @@
+fetch('products.json')
+  .then((response) => response.json())
+  .then((data) => {
+    const products = data.products;
+
+    let productListHTML = '';
+
+    products.forEach(product => {
+      productListHTML += `
+        <div>
+          <span>${product.name} - $${product.price}</span>
+          <button class="add-to-cart" data-product-id="${product.id}">Add to Cart</button>
+        </div>
+      `;
+    });
+
+    const productListContainer = document.getElementById('product-list');
+    productListContainer.innerHTML = productListHTML;
+
+    const addToCartButtons = document.querySelectorAll('.add-to-cart');
+    addToCartButtons.forEach(button => {
+      button.addEventListener('click', addToCart);
+    });
+
+    function addToCart(event) {
+      const productId = event.target.dataset.productId;
```

Comment:
```suggestion
    function addToCart({ target { dataset: { productId }}}) {
```

Não testei, mas algo nessa linha deve funcionar.